content/guides/hot-module-replacement.md

```diff
@@ -18,140 +19,271 @@ related:
+  import _ from 'lodash';
+  import printMe from './print.js';
 
-if (module.hot) {
-  module.hot.accept('./library', function() {
-    console.log('Accepting the updated library module!');
-    Library.log();
-  })
-}
++ if (module.hot) {
++   module.hot.accept('./print.js', function() {
++     console.log('Accepting the updated printMe module!');
++     printMe();
++   })
++ }
+
+  function component() {
+    var element = document.createElement('div');
+    var btn = document.createElement('button');
+
+    element.innerHTML = _.join(['Hello', 'webpack'], ' ');
+
+    btn.innerHTML = 'Click me and check the console!';
+    btn.onclick = printMe;
+
+    element.appendChild(btn);
+
+    return element;
+  }
+
+  document.body.appendChild(component());
 ```
 
-__library.js__
+Start changing the `console.log` statement in `print.js`, and you should see the following output in the browser console.
 
-``` js
-export default {
-  log() {
-    // Change this after the server is started to test
-    console.log('Initial log...')
+__print.js__
+
+``` diff
+  export default function printMe() {
+-   console.log('I get called from print.js!');
++   console.log('Updating print.js...')
   }
-}
 ```
 
-Start changing the `console.log` statement in `library.js`, to `'Second log...'` for example, and you should see the following output in the browser console...
+__console__
 
 ``` diff
 [HMR] Waiting for update signal from WDS...
-main.js:9998 Initial log...
-main.js:9468 [WDS] Hot Module Replacement enabled.
-+ 2main.js:9468 [WDS] App updated. Recompiling...
-+ main.js:9468 [WDS] App hot update...
-+ main.js:9912 [HMR] Checking for updates on the server...
-+ main.js:9982 Accepting the updated library module!
-+ 0.1bafc70….hot-update.js:11 Second log...
-+ main.js:9955 [HMR] Updated modules:
-+ main.js:9957 [HMR]  - ./src/library.js
-+ main.js:9894 [HMR] App is up to date.
+main.js:4395 [WDS] Hot Module Replacement enabled.
++ 2main.js:4395 [WDS] App updated. Recompiling...
++ main.js:4395 [WDS] App hot update...
++ main.js:4330 [HMR] Checking for updates on the server...
++ main.js:10024 Accepting the updated printMe module!
++ 0.4b8ee77….hot-update.js:10 Updating print.js...
++ main.js:4330 [HMR] Updated modules:
++ main.js:4330 [HMR]  - 20
++ main.js:4330 [HMR] Consider using the NamedModulesPlugin for module names.
 ```
 
 
 ## Gotchas
 
-Hot Module Replacement can be tricky. For example, let's say I have the following class:
+Hot Module Replacement can be tricky. To show this, let's go back to our working example. If you go ahead and click the button on the example page, you will realize the console is printing the old `printMe` function.
+
+This is happening because the button's `onclick` event handler is still bound to the original `printMe` function.
 
-``` js
-class Logger {
-  log(text) {
-    console.log('Logging some text: ', text)
+To make this work with HMR we need to update that binding to the new `printMe` function using `module.hot.accept`
```

Comment:
You should probably to add some punctuation to the end of this sentence (e.g. `:`).